compiler/rustc_borrowck/src/diagnostics/conflict_errors.rs

```diff
@@ -363,11 +363,8 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
             }
         }
         let hir = self.infcx.tcx.hir();
-        if let Some(hir::Node::Item(hir::Item {
-            kind: hir::ItemKind::Fn(_, _, body_id),
-            ..
-        })) = hir.find(self.mir_hir_id())
-            && let Some(hir::Node::Expr(expr)) = hir.find(body_id.hir_id)
+        if let Some(body_id) = hir.maybe_body_owned_by(self.mir_def_id()) &&
+            let Some(hir::Node::Expr(expr)) = hir.find(body_id.hir_id)
         {
```

Comment:
A bit more canonical:
```suggestion
        if let Some(body_id) = hir.maybe_body_owned_by(self.mir_def_id()) {
            let expr = hir.body(body_id).value;
```